owncloudData/src/test/java/com/owncloud/android/data/files/repository/OCFileRepositoryTest.kt

```diff
@@ -25,382 +26,2010 @@ import com.owncloud.android.data.files.datasources.LocalFileDataSource
+import kotlinx.coroutines.flow.first
 import kotlinx.coroutines.flow.flowOf
 import kotlinx.coroutines.test.runTest
 import org.junit.Assert.assertEquals
 import org.junit.Assert.assertNull
-import org.junit.Ignore
+import org.junit.Assert.assertThrows
+import org.junit.Before
 import org.junit.Test
+import java.util.UUID
 
-@Ignore("Ignore temporary, pretty dependant on implementation... Will be reworked when finished")
 @ExperimentalCoroutinesApi
 class OCFileRepositoryTest {
 
-    private val remoteFileDataSource = mockk<RemoteFileDataSource>(relaxed = true)
-    private val localFileDataSource = mockk<LocalFileDataSource>(relaxed = true)
-    private val localSpacesDataSource = mockk<LocalSpacesDataSource>(relaxed = true)
-    private val localStorageProvider = mockk<LocalStorageProvider>()
-    private val ocFileRepository: OCFileRepository =
-        OCFileRepository(localFileDataSource, remoteFileDataSource, localSpacesDataSource, localStorageProvider)
-
-    private val folderToFetch = OC_FOLDER
-    private val listOfFilesRetrieved = listOf(
-        folderToFetch,
-        OC_FOLDER.copy(remoteId = "one"),
-        OC_FOLDER.copy(remoteId = "two")
-    )
-    private val listOfFileToRemove = listOf(
-        OC_FOLDER.copy(id = 1),
-        OC_FILE.copy(id = 2),
-        OC_FILE.copy(id = 3)
+    private val localFileDataSource = mockk<LocalFileDataSource>(relaxUnitFun = true)
+    private val remoteFileDataSource = mockk<RemoteFileDataSource>(relaxUnitFun = true)
+    private val localSpacesDataSource = mockk<LocalSpacesDataSource>(relaxUnitFun = true)
+    private val localStorageProvider = mockk<LocalStorageProvider>(relaxUnitFun = true)
+    private val ocFileRepository = OCFileRepository(
+        localFileDataSource,
+        remoteFileDataSource,
+        localSpacesDataSource,
+        localStorageProvider
     )
+    private val ocFileRepositorySpy = spyk(ocFileRepository)
+
+    private val expectedRemotePath = OC_FOLDER_WITH_SPACE_ID.remotePath + OC_FILE_WITH_SPACE_ID.fileName
+    private val storagePath = "/local/storage/path/username@demo.owncloud.com/Folder/Photos/image2.jpt"
+    private val remoteId = "remoteId"
+    private val searchText = "image"
+    private val etagInConflict = "5efb0c13c688i"
+    private val fileWithConflict = OC_FILE_WITH_SPACE_ID.copy(etagInConflict = etagInConflict)
+
+    @Before
+    fun setUp() {
+        val commonSpaceId = OC_FILE_WITH_SPACE_ID.spaceId
+        val commonAccountName = OC_FILE_WITH_SPACE_ID.owner
+        every {
+            localSpacesDataSource.getWebDavUrlForSpace(
+                spaceId = commonSpaceId,
+                accountName = commonAccountName
+            )
+        } returns OC_SPACE_PERSONAL.root.webDavUrl
+    }
+
+    @Test
+    fun `createFolder creates a new folder and saves it`() {
+        every {
+            localSpacesDataSource.getWebDavUrlForSpace(
+                spaceId = OC_PARENT_FOLDER_WITH_SPACE_ID.spaceId,
```

Comment:
This is used on different places, what do you think to create a generic structure as 

private fun mockWebDavUrlForSpace(spaceId: String, owner: String, url: String) {
    every {
        localSpacesDataSource.getWebDavUrlForSpace(spaceId, owner)
    } returns url
}

and reuse it on different tests?

Reply:
As long as I can see, this is only used twice in this class in `every` blocks (one in `setUp()` method and another one here), the other uses are in `verify` blocks, which need to contain the method itself. But, it's true that since both uses in `every` blocks get the same parameters, this one can be removed and keep just the one in `setUp()`, so that we don't need to do a generic method 👍